src/shmem_internal.h

```diff
@@ -598,7 +598,8 @@ static inline
 int shmem_internal_pe_in_active_set(int global_pe, int PE_start, int PE_stride, int PE_size)
 {
     int n = (global_pe - PE_start) / PE_stride;
-    if (global_pe < PE_start || (global_pe - PE_start) % PE_stride || n >= PE_size)
+    if ((global_pe < PE_start && PE_stride > 0) || (global_pe > PE_start && PE_stride < 0) ||
+        (global_pe - PE_start) % PE_stride || n >= PE_size)
```

Comment:
If `PE_stride` is negative, then `n` can become negative. Then, the last comparison `n >= PE_size` will never be true, because PE_size cannot be negative. I am trying to come with an example where this causes a problem, but could not. May be this is correct.

Reply:
If `PE_stride` is negative, then `n` is only negative if `global_pe` > `PE_start`.  But in that case, `global_pe` would _not_ be in the active set (because `PE_start` has the largest PE value in the set) so `n >= PE_size` is never checked (because the 2nd or'd condition is true).  So I think this is ok.